source/validate_id.cpp

```diff
@@ -181,6 +181,27 @@ bool idUsage::isValid<SpvOpMemberDecorate>(const spv_instruction_t* inst,
 }
 
 template <>
+bool idUsage::isValid<SpvOpDecorationGroup>(const spv_instruction_t* inst,
+                                     const spv_opcode_desc) {
+  auto decorationGroupIndex = 1;
+  auto decorationGroup = module_.FindDef(inst->words[decorationGroupIndex]);
+
+  for (auto pair : decorationGroup->uses()) {
+    auto use = pair.first;
+    if (use->opcode() != SpvOpDecorate &&
+        use->opcode() != SpvOpGroupDecorate &&
+        use->opcode() != SpvOpGroupMemberDecorate &&
+        use->opcode() != SpvOpName ) {
+      DIAG(decorationGroupIndex) << "Result id of OpDecorationGroup can only "
```

Comment:
It would be more friendly to give more information so we can identify the problem in a big SPIR-V module:
- give the result ID of the OpDecorationGroup (so  you can easily find it)
- give the Opcode of the offending instruction.  Optional: If it has a result Id then give that too.

Generally, the validator's error messages are pretty poor.  I'd like to incrementally make them better.